lib/test.js

```diff
@@ -134,7 +138,11 @@ Test.prototype.run = function () {
 	}.bind(this));
 };
 
-Test.prototype.end = function () {
+Test.prototype.end = function (err) {
+	if (arguments.length >= 1 && err) {
```

Comment:
Hmmm.

We need it to behave consistently whether or not it's being used as a callback

```js
t.plan(1);
t.ok(true);
t.end();
```

should perform the same as

```js
t.plan(1)
t.ok(true);
setImmediate(t.end);
```